nvflare/fuel/f3/mpm.py

```diff
@@ -139,28 +139,31 @@ def run(cls, main_func, shutdown_grace_time=1.5, cleanup_grace_time=1.5):
 
         # call and wait for the main_func to complete
         logger = cls.logger()
-        logger.debug(f"=========== {cls.name}: started to run forever")
-        rc = main_func()
-
-        # start shutdown process
-        cls._stopping = True
-        cls._start_shutdown(shutdown_grace_time, cleanup_grace_time)
-
-        # We can now stop the AIO loop!
-        AioContext.close_global_context()
-
-        logger.debug(f"=========== {cls.name}: checking running threads")
-        num_active_threads = 0
-        for thread in threading.enumerate():
-            if thread.name != "MainThread" and not thread.daemon:
-                logger.warning(f"#### {cls.name}: still running thread {thread.name}")
-                num_active_threads += 1
-
-        logger.info(f"{cls.name}: Good Bye!")
-        if num_active_threads > 0:
-            try:
-                os.kill(os.getpid(), signal.SIGKILL)
-            except Exception as ex:
-                logger.debug(f"Failed to kill process {os.getpid()}: {secure_format_exception(ex)}")
+        logger.debug(f"=========== {cls.name}: starting main_func")
+        try:
```

Comment:
I think Yan is making some changes to the MPM shutting down process. Could you please sync with him on the changes?

Reply:
Ok will do

Reply:
I will disable the changes here as #1597 is changing MPM.py